data-prepper-plugins/parquet-codecs/src/main/java/org/opensearch/dataprepper/plugins/codec/parquet/DataTypeChecker.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.codec.parquet;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.parquet.example.data.simple.SimpleGroup;
+import org.apache.parquet.schema.PrimitiveType;
+
+import org.apache.parquet.schema.LogicalTypeAnnotation;
+import org.apache.parquet.schema.Type;
+
+import java.util.UUID;
+
+public class DataTypeChecker {
+
+    private DataTypeChecker(){
+
+    }
+
+    public static Object checkDataType(Type field, SimpleGroup simpleGroup, int fieldIndex) throws Exception {
+        String fieldPrimitiveDataType;
+        if (field.isPrimitive()) {
+            PrimitiveType primitiveType = field.asPrimitiveType();
+            fieldPrimitiveDataType = primitiveType.getPrimitiveTypeName().name();
+            switch (fieldPrimitiveDataType) {
+                case "INT32":
+                    return simpleGroup.getInteger(fieldIndex, 0);
+                case "BOOLEAN":
+                    return simpleGroup.getBoolean(fieldIndex, 0);
+                case "FLOAT":
+                    return simpleGroup.getFloat(fieldIndex, 0);
+                case "DOUBLE":
+                    return simpleGroup.getDouble(fieldIndex, 0);
+                case "INT96":
+                    return simpleGroup.getInt96(fieldIndex, 0);
+                case "INT64":
+                case "LONG":
+                    return simpleGroup.getLong(fieldIndex,0);
+                case "FIXED_LEN_BYTE_ARRAY":
+                    return simpleGroup.getValueToString(fieldIndex, 0);
+                default:
+                    throw new RuntimeException("The Parquet Codec doesn't support this data-type yet.");
```

Comment:
Let's can make this error message more useful for users by including the type name. 

```
throw new RuntimeException("The Parquet codec doesn't support the '" + fieldPrimitiveDataType + "' data type.");
```

I made a few other tweaks.

Reply:
Sure.